### PR TITLE
Add GroupBy.apply()

### LINKED ISSUE #409: DataFrame.GroupBy.apply()
I'm continuously missing this behavior when converting my pandas scripts to koalas:

```python
df = pd.DataFrame({"timestamp":[0.0, 0.5, 1.0, 0.0, 0.5],
              "car_id": ['A','A','A','B','B'], 
              "battery_charge": [100, 90, 80,100,90]
              })

print(df)

def calc_battery_usage(x):
    start_bat = x.sort_values('timestamp').iloc[0]['battery_charge']
    stop_bat = x.sort_values('timestamp').iloc[-1]['battery_charge']
    return start_bat - stop_bat

print(df.groupby('car_id').apply(calc_battery_usage))
```
That prints out:
```python
   timestamp car_id  battery_charge
0        0.0      A             100
1        0.5      A              90
2        1.0      A              80
3        0.0      B             100
4        0.5      B              90

car_id
A    20
B    10
```

On koalas I just get:
```
PandasNotImplementedError: The method `pd.groupby.GroupBy.apply()` is not implemented yet.
```

I can take over and start implementing it myself, but would probably need some general guidelines on how to do it (probably using pysparks pandas_udf).

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR adds `GroupBy.apply` by using grouped map pandas UDF.

Resolves #409

### CODE REVIEW

File: databricks/koalas/generic.py [hunk 1]
```diff
@@ -929,7 +929,7 @@ def skew(self, numeric_only=True):
 
         On a DataFrame:
 
-        >>> df.skew()
+        >>> df.skew()  # doctest: +SKIP
```
Comment (HyukjinKwon) [inline databricks/koalas/generic.py]: This change is not related but seems this test is flaky for some reasons.

Comment (HyukjinKwon) [reply databricks/koalas/generic.py]: ` b   -3.319678e-16` isn't exactly matched sometimes.

Comment (HyukjinKwon) [reply databricks/koalas/generic.py]:
```
     a    0.000000e+00
    -b   -3.319678e-16
    +b   -2.655742e-16
```

**Comment by HyukjinKwon** [comment]
Tests passed

**Comment by HyukjinKwon** [comment]
Tests passed

File: databricks/koalas/groupby.py [hunk 2]
```diff
@@ -340,6 +344,103 @@ def any(self):
...
+            # For now, just positionally map the column names to given schema's.
+            pdf = pdf.rename(columns=dict(zip(pdf.columns, return_schema.fieldNames())))
+            return pdf
+
+        grouped_map_func = pandas_udf(return_schema, PandasUDFType.GROUPED_MAP)(rename_output)
+
+        sdf = self._kdf._sdf
+        data_columns = self._kdf._internal.data_columns
+        input_groupkeys = [s._scol for s in self._groupkeys]
+        sdf = sdf.select(data_columns).groupby(*input_groupkeys).apply(grouped_map_func)
```
Comment (ueshin) [inline databricks/koalas/groupby.py]: Do we need to `select(data_columns)` here? If needed we should use `data_scols` instead.

File: databricks/koalas/groupby.py [hunk 3]
```diff
@@ -340,6 +344,103 @@ def any(self):
...
+        See Also
+        --------
+        aggregate : Apply aggregate function to the GroupBy object.
+        Series.apply : Apply a function to a Series.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'A': 'a a b'.split(),
+        ...                    'B': [1,2,3],
+        ...                    'C': [4,6, 5]}, columns=['A', 'B', 'C'])
```
Comment (ueshin) [inline databricks/koalas/groupby.py]: nit: the values for `C` has an extra space, or rather need extra spaces?

**Pull request merged**